lib/Echidna/Processor.hs

```diff
@@ -0,0 +1,113 @@
+
+module Echidna.Processor where
+
+import Control.Arrow              (second)
+import Control.Lens
+import Control.Monad.IO.Class     (MonadIO(..))
+import Control.Exception          (Exception)
+import Control.Monad.Catch        (MonadThrow(..))
+import Data.Aeson                 (decode, Value(..))
+import System.Directory           (findExecutable)
+import System.Process             (StdStream(..), readCreateProcessWithExitCode, proc, std_err)
+import System.Exit                (ExitCode(..))
+import GHC.Word                   (Word32)
+
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import qualified Data.HashMap.Strict as M
+import qualified Data.Text           as T
+
+import Echidna.ABI (hashSig)
+
+-- | Things that can go wrong trying to run a processor. Read the 'Show'
+-- instance for more detailed explanations.
+data ProcException = ProcessorFailure String String
+                   | ProcessorNotFound String
+
+instance Show ProcException where
+  show = \case
+    ProcessorFailure p e     -> "Error running " ++ p ++ ": " ++ e
+    ProcessorNotFound p      -> "Cannot find processor " ++ p
+
+instance Exception ProcException
+
+{-
+process :: (MonadIO m, MonadThrow m) => FilePath -> Maybe String -> EConfig -> m EConfig
+process f c e = do 
+                 r <- runSlither f (e ^. sConf ^. cryticArgs)
+                 --liftIO $ print $ r
+                 --liftIO $ print $ filterResults c $ filterPayable r
+                 --liftIO $ print $ filterResults c $ filterConstantFunction x
+                 --let ps = concatMap (map hashSig . snd) rs
+                 return $ e & xConf . payableSigs .~ filterResults c (filterPayable r)
+-}
+
+filterResults :: Maybe String -> [(T.Text, [T.Text])] -> [Word32] 
+filterResults (Just c) rs = case lookup (T.pack c) rs of
+                             Nothing -> filterResults Nothing rs
+                             Just s -> map hashSig s 
+filterResults Nothing rs = concatMap (map hashSig . snd) rs
+
+data SlitherInfo = PayableInfo (T.Text, [T.Text]) 
+                 | ConstantFunctionInfo (T.Text, [T.Text]) deriving (Show)
+
+filterPayable :: [SlitherInfo] -> [(T.Text, [T.Text])]
+filterPayable = map g . filter f
+                 where f (PayableInfo _) = True
+                       f _               = False
+                       g (PayableInfo i) = i
+                       g _               = error "fail in filterPayable"
+
+filterConstantFunction :: [SlitherInfo] -> [(T.Text, [T.Text])]
+filterConstantFunction = map g . filter f 
+                 where f (ConstantFunctionInfo _) = True
+                       f _               = False
+                       g (ConstantFunctionInfo i) = i
+                       g _               = error "fail in filterPayable"
+
+-- Slither processing
+runSlither :: (MonadIO m, MonadThrow m) => FilePath -> [String] -> m [SlitherInfo]
+runSlither fp aa = let args = ["--print", "echidna", "--json", "-"] ++ aa in do
```

Comment:
This will run the compilation a second time, which can take some time. 
Some ideas:
- run Slither with `--ignore-compile` flag
- do not run `crytic-compile` at all, and add the ast in the printer result

Reply:
Sorry, I forgot to update this here. I added the `--ignore-compile` flag so slither remains optional. We will have a better integration on the following release (and warn the user about the mandatory use of slither).